drivers/can/mcp2515.c

```diff
@@ -0,0 +1,748 @@
+	struct k_sem int_sem;
+
+	/* tx data */
+	struct k_sem tx_sem;
+	struct k_mutex tx_mutex;
+	struct mcp2515_tx_cb tx_cb[MCP2515_TX_CNT];
+	u8_t tx_busy_map;
+
+	/* filter data */
+	struct k_mutex filter_mutex;
+	u64_t filter_usage;
+	u64_t filter_response_type;
+	void *filter_response[CONFIG_CAN_MAX_FILTER];
+	struct can_filter filter[CONFIG_CAN_MAX_FILTER];
+};
+
+struct mcp2515_config {
+	/* spi configuration */
+	const char *spi_port;
+	u8_t spi_cs_pin;
+	const char *spi_cs_port;
+	u32_t spi_freq;
+	u8_t spi_slave;
+
+	/* interrupt configuration */
+	u8_t int_pin;
+	const char *int_port;
+	size_t int_thread_stack_size;
+	int int_thread_priority;
+
+	/* CAN timing */
+	u8_t tq_sjw;
+	u8_t tq_prop;
+	u8_t tq_bs1;
+	u8_t tq_bs2;
+};
+
+/* MCP2515 Opcodes */
+#define MCP2515_OPCODE_WRITE            0x02
+#define MCP2515_OPCODE_READ             0x03
+#define MCP2515_OPCODE_BIT_MODIFY       0x05
+#define MCP2515_OPCODE_READ_STATUS      0xA0
+#define MCP2515_OPCODE_RESET            0xC0
+
+/* MCP2515 Registers */
+#define MCP2515_ADDR_CANSTAT            0x0E
+#define MCP2515_ADDR_CANCTRL            0x0F
+#define MCP2515_ADDR_CNF3               0x28
+#define MCP2515_ADDR_CNF2               0x29
+#define MCP2515_ADDR_CNF1               0x2A
+#define MCP2515_ADDR_CANINTE            0x2B
+#define MCP2515_ADDR_CANINTF            0x2C
+#define MCP2515_ADDR_TXB0CTRL           0x30
+#define MCP2515_ADDR_TXB1CTRL           0x40
+#define MCP2515_ADDR_TXB2CTRL           0x50
+#define MCP2515_ADDR_RXB0CTRL           0x60
+#define MCP2515_ADDR_RXB1CTRL           0x70
+
+/* MCP2515 Operation Modes */
+#define MCP2515_MODE_NORMAL             0x00
+#define MCP2515_MODE_LOOPBACK           0x02
+#define MCP2515_MODE_SILENT             0x03
+#define MCP2515_MODE_CONFIGURATION      0x04
+
+/* MCP2515_FRAME_OFFSET */
+#define MCP2515_FRAME_OFFSET_SIDH       0
+#define MCP2515_FRAME_OFFSET_SIDL       1
+#define MCP2515_FRAME_OFFSET_EID8       2
+#define MCP2515_FRAME_OFFSET_EID0       3
+#define MCP2515_FRAME_OFFSET_DLC        4
+#define MCP2515_FRAME_OFFSET_D0         5
+
+/* MCP2515_CANINTF */
+#define MCP2515_CANINTF_RX0IF           BIT(0)
+#define MCP2515_CANINTF_RX1IF           BIT(1)
+#define MCP2515_CANINTF_TX0IF           BIT(2)
+#define MCP2515_CANINTF_TX1IF           BIT(3)
+#define MCP2515_CANINTF_TX2IF           BIT(4)
+#define MCP2515_CANINTF_ERRIF           BIT(5)
+#define MCP2515_CANINTF_WAKIF           BIT(6)
+#define MCP2515_CANINTF_MERRF           BIT(7)
+
+static int mcp2515_soft_reset(struct device *dev)
+{
+	u8_t opcode_buf[1] = { MCP2515_OPCODE_RESET };
+	const struct spi_buf tx_buf = {
+		.buf = opcode_buf,
+		.len = 1,
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+
+	return spi_write(DEV_DATA(dev)->spi, &DEV_DATA(dev)->spi_cfg, &tx);
+}
+
+static int mcp2515_write_reg(struct device *dev, u8_t reg_addr, u8_t *buf_data,
+			     u8_t buf_len)
+{
+	u8_t opcode_buf[2];
+
+	opcode_buf[0] = MCP2515_OPCODE_WRITE;
```

Comment:
rewrite that: 

u8_t opcode_buf = { ..., ... };

Apply that comment everywhere. (you did it properly in soft-reset function above actually)